src/tunnel.cc

```diff
@@ -1016,11 +1082,16 @@ void
 TunnelStateData::connectedToPeer(Security::EncryptorAnswer &answer)
 {
     if (ErrorState *error = answer.error.get()) {
-        answer.error.clear(); // sendError() will own the error
-        sendError(error, "TLS peer connection error");
-        return;
+        saveError(error);
+        answer.error.clear();
+        Must(server.monitorsClosures()); // closure notification (is already pending or) will happen
+        if (IsConnOpen(server.conn))
+            server.conn->close();
+        return; // tunnelServerClosed() will retry as needed
     }
 
+    retriable = false;
```

Comment:
Should not TunnelStateData retry Http::Tunneler failures? FwdState does retry them, right?

Reply:
Done.